docs: Review Create New Section doc (#1874)
vtex/faststore: apps/site/pages/docs/building-sections/creating-a-new-section.mdx

  This documentation is currently under development.
</Callout>

> In this guide, you'll learn how to create a new section for your store and make it available in the VTEX Headless CMS. This solution is useful when your store requires a section not natively available in FastStore.

Reviewer: ```suggestion
In this guide, you'll learn how to create a new section for your store and make it available in the [VTEX Headless CMS](/docs/headless-cms-overview). This solution is useful when your store requires a section not natively available in FastStore.
```

For this guide we'll create a `Call to Action` section.

<Callout type="info" emoji="ℹ️">
  Sections are page components that wrap up other components, enabling you to create cohesive and functional content for your store. For example, the Hero section is a native section that includes the
  following components: `Hero`, `HeroImage`, and `HeroHeader`.
</Callout>

---

## Before you start

**1. Integrate the store with the VTEX Headless CMS**

All sections must be available in the Headless CMS so they can be added and managed on your store's pages. To integrate your FastStore project, please refer to the [VTEX Headless CMS integration track](/docs/headless-cms-integration/1-configuring-the-vtex-account).

**2. Knowledge on how sections and content types work on Headless CMS**

> During the creation of a new section, you will create files such as `sections.json` and `content-types.json`, which follow a structure established for the Headless CMS.

Reviewer: ```suggestion
During the creation of a new section, you will create files such as `sections.json` and `content-types.json`, which follow a structure established for the VTEX Headless CMS.
```

1. In the FastStore root directory, create a folder named `cms`.
2. Inside `cms`, create the `faststore` folder.
3. Within the `cms/faststore` folder, create the following files: `content-types.json` and `sections.json`.
4. In the `sections.json` file add the new section that you want to display in the Headless CMS. The schema below defines how the Headless CMS renders a section:

```json filename="sections.json"
[
  {
    "name": "CallToAction",
    "schema": {
      "title": "Call To Action",
      "description": "Get your 20% off on the first purchase!",
      "type": "object",
      "required": [
        "title",
        "link"
      ],
      "properties": {
        "title": {
          "title": "Title",
          "type": "string"
        },
        "link": {
          "title": "Link Path",
          "type": "object",
          "required": [
            "text",
            "url"
          ],
          "properties": {
            "text": {
              "title": "Text",
              "type": "string"
            },
            "url": {
              "title": "URL",
              "type": "string"
            }
          }
        }
      }
    }
  }
]
```

> This new section receives a title and a link pointing to somewhere.

Reviewer: ```suggestion
<Callout type="info" emoji="ℹ️">
  This new section receives a title and a link pointing to a specific location.
</Callout>
```

### Step 2: Creating the new section

To render the `CallToAction` section you created in the previous step, you need to create this section component.

1. If you don't already have it, create a folder named `components` inside the `src` directory.
2. Inside the `components` folder, create a file named `CallToAction.tsx` and add the following code:

> ```tsx filename="components/CallToAction.tsx"

Author: ```suggestion
```tsx filename="src/components/CallToAction.tsx"
```

import React from "react";

export interface CallToActionProps {
  title: string;
  link: {
    text: string;
    url: string;
  };
}

export default function CallToAction(props: CallToActionProps) {
  return (
    <section>
      <h2>{props.title}</h2>
      <a href={props.link.url}>{props.link.text}</a>
    </section>
  );
}
```

This section will receive the `link` and `title` defined previously in the `sections.json` file.

> 3. Create a file named `index.tsx` inside the components folder.

Author: ```suggestion
3. Create a file named `index.ts` inside the components folder.
```

Open the index.tsx file and add the following code:

```tsx filename="components/index.tsx"
import CallToAction from './CallToAction'

export default { CallToAction }
```

### Step 3: Synchronizing the new section with the Headless CMS

1. In the terminal, run `faststore dev`.
2. Then, run `faststore cms-sync`. This command will synchronize the new section you created with the CMS.
2. Go to the VTEX Admin and access **Storefront > Headless CMS**.
3. Click on the `Home` content type.

> 4. In the `Sections` tab, click the `+`, search for the new `Call to Action` section, and add it to your page.

Author: Add image here.

> Open the index.tsx file and add the following code:

Author: ```suggestion
Open the `index.ts` file and add the following code:
```

> ```tsx filename="components/index.tsx"

Author: ```suggestion
```tsx filename="components/index.ts"
```
